GillianCore/logging/logging.mli

```diff
@@ -94,6 +94,14 @@ module LogQueryer : sig
   val get_unify_for : ReportId.t -> (ReportId.t * string) option
 end
 
+type report_state
+
+val new_report_state : unit -> report_state
+val clone_report_state : report_state -> report_state
+val global_report_state : report_state
+val activate_report_state : report_state -> unit
+val with_report_state : (unit -> 'a) -> report_state -> 'a
+
```

Comment:
```ocaml
module Report_state = struct 
  type t
  val make: unit -> t
  val copy : t -> t
  val global : t
  val activate : t -> unit
  val with_report_state: (unit -> 'a) -> report_state -> 'a
end
```
or something similar? Whatever makes sense? Maybe do that in `ReportBuilder` directly, and then just alias the module?

Reply:
Resolved in 29a513adb40f0d75f819d837419964f12f3c370c